### PR TITLE
Add additional pre-commit hooks

### PR DESCRIPTION
**Pull request by mattkram**
Adds additional pre-commit hooks related to whitespace & formatting of `.yaml` files.

Each hook is added and applied in a single atomic commit.

### CODE REVIEW

File: pyscriptjs/environment.yml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 channels:
-  - defaults
-  - conda-forge
+-   defaults
```
Comment (verhulstm) [inline pyscriptjs/environment.yml:2]: why this formatting change?

**Review by verhulstm — APPROVED** [review]
i dont see any issues

File: pyscriptjs/environment.yml (hunk 1 above)
Comment (mattkram) [reply pyscriptjs/environment.yml:2]: It's from the auto-formatter. This one aligns list items with the parent. Indentation of the text is configurable. If eventually there is consensus that we'd prefer it differently, it wouldn't be tough to fork or contribute a new config option to the parent hook.

**Review by fpliger — APPROVED** [review]
Awesome! Once we fix the conflicts it can be merged

**Comment by mattkram** [comment]
🚀

**Pull request merged**